### ISSUE TITLE
Is Faker integration easy/possible? Any tips on how?

### ISSUE THREAD

**Issue by dave-cross** [pattern-lab/edition-php-twig-standard#7]
I see a older thread in the php edition that talked about adding _Faker_. Did that ever happen?

Quick research says Faker can be autoloaded in _Symfony_ projects… might make a good addition to the _Twig_ version of **PL.** I don't have the experience or knowledge to know how.

If it can be added into **PL-twig** easily or someone could give me the quick rundown of how to add it myself, I'd appreciate it.

**Comment by dave-cross** [comment]
Learned some things and found a way. Documenting here for others.

Add the Faker to your project with:
`composer require fzaninotto/faker`

Create a new function in `source/_twig_components/functions/` called _faker.function.php_
Copy this code into it:

```
<?php

$function = new Twig_SimpleFunction('faker', function ($string, $options="") {
  $faker = Faker\Factory::create();
  if ($options != "") {
    return $faker->$string($options);
  } else {
    return $faker->$string;
  }
});

?>
```

Call it in your Twig files with:

```
{{ faker('PROPERTY YOU WANT', 'options') }} {# `options` are optional #}
{{ faker('name') }} {# returns 'Neal Considine' #}
{{ faker('firstname') }} {# returns 'Jaron' #}
{{ faker('lastname') }} {# returns 'Will' #}
{{ faker('paragraph') }} {# returns paragraph #}
{{ faker('text', '500') }} {# returns 500 words #}
{{ faker('words', '5')|join(' ') }} {# returns array of words. use `join` filter #}
```

I haven't tested with all the various properties, but the ones I have tried work.

Improvements and advice humbly accepted.

-Dave

**Comment by dmolsen** [comment]
Wow. That's cool :) Faker support has been on the list for a very long time. This is a really smart way of adding support. I'll try to find a more permanent home to document this.

**Comment by danny-englander** [comment]
I am really glad I found this post, thanks for all the info. Faker is amazing and I've been looking for a way to streamline our workflow. Faker seems to fit the bill. I discovered I can create a loop of data like:

```
 {% for i in range(0, 8) | slice(1, 4) %}
        <p>{{ faker('firstname') }} {{ faker('lastname') }}</p>
        <p>{{ faker('paragraph') }}</p>
        <img src="{{ faker('imageUrl', '360x210') | replace({'http': "https", "gif" : "jpg"}) }}">
        <hr>
    {% endfor %}
```

In addition, I added [placholdit](https://github.com/EmanueleMinotto/PlaceholdItProvider) as the default image provider as Faker now has an API for things like that. So after installing that, `imageUrl` defaults to placeholdit which is what we were using / manually writing in our JSON files before Faker.
